applications/sim_showers_for_trigger_rates.py

```diff
@@ -0,0 +1,186 @@
+
+        python applications/sim_showers_for_trigger_rates.py -a 4LST -s North \
+        --primary proton --nruns 100 --nevents 10000 --output {some dir for large files}
+'''
+
+import logging
+import argparse
+
+import astropy.units as u
+
+import simtools.io_handler as io
+import simtools.config as cfg
+import simtools.util.general as gen
+from simtools.shower_simulator import ShowerSimulator
+
+
+if __name__ == '__main__':
+
+    parser = argparse.ArgumentParser(
+        description=(
+            'Calculate and plot the PSF and eff. mirror area as a function of off-axis angle '
+            'of the telescope requested.'
+        )
+    )
+    parser.add_argument(
+        '-a',
+        '--array',
+        help='Name of the array (e.g. 1MST, 4 LST ...)',
+        type=str,
+        required=True
+    )
+    parser.add_argument(
+        '-s',
+        '--site',
+        help='Site name (North or South)',
+        type=str,
+        required=True
+    )
+    parser.add_argument(
+        '--primary',
+        help='Name of the primary particle (e.g. proton, helium ...)',
+        type=str,
+        required=True
+    )
+    parser.add_argument(
+        '--nruns',
+        help='Number of runs (default=100)',
+        type=int,
+        default=100
+    )
+    parser.add_argument(
+        '--nevents',
+        help='Number of events/run (default=100)',
+        type=int,
+        default=100000
+    )
+    parser.add_argument(
+        '--zenith',
+        help='Zenith angle in deg (default=20)',
+        type=float,
+        default=20
+    )
+    parser.add_argument(
+        '--azimuth',
+        help='Azimuth angle in deg (default=0)',
+        type=float,
+        default=0
+    )
+    parser.add_argument(
+        '--output',
+        help='Path of the output directory where the simulations will be saved.',
+        type=str,
+        default=None
+    )
+    parser.add_argument(
+        '--test',
+        help='Test option will not submit any job.',
+        action='store_true'
+    )
+    parser.add_argument(
+        '-v',
+        '--verbosity',
+        dest='logLevel',
+        action='store',
+        default='info',
+        help='Log level to print (default is INFO)'
+    )
+
+    args = parser.parse_args()
+    label = 'trigger_rates'
+
+    logger = logging.getLogger()
+    logger.setLevel(gen.getLogLevelFromUser(args.logLevel))
+
+    # Output directory to save files related directly to this app
+    outputDir = io.getApplicationOutputDirectory(cfg.get('outputLocation'), label)
+
+    showerConfigData = {
+        'corsikaDataDirectory': args.output,
+        'site': args.site,
+        'layoutName': args.array,
+        'runRange': [1, args.nruns + 1],
+        'nshow': args.nevents,
+        'primary': args.primary,
+        'erange': [100 * u.GeV, 300 * u.TeV],
+        'eslope': -2,
+        'zenith': args.zenith * u.deg,
+        'azimuth': args.azimuth * u.deg,
+        'viewcone': 10 * u.deg,
```

Comment:
Is viewcone always fixed? Does that depend on the FOV of the considered telescope type?

Reply:
I'm following what has been done by Konrad. You can see the file corsika-run/INPUTS_CTA-trg-test-prod3-LST4_proton, for example.
I don't actually know whether is a good idea to reduce the FoV or not. It's easy to do anyway.

The change wouldn't be very large anyway, right?

Reply:
I am still trying to understand exactly: all parameters given as values in this dict are fixed and not changeable by the user?

Having a too large FOV is a waste of computing time. Why would one fix the energy range? 10 GeV is correct for LSTs, but definitely far too low for SSTs.

Reply:
> I am still trying to understand exactly: all parameters given as values in this dict are fixed and not changeable by the user?

Yes. But note that this is a specific application to produce simulations for trigger rate calculations.  

> Having a too large FOV is a waste of computing time. Why would one fix the energy range? 10 GeV is correct for LSTs, but definitely far too low for SSTs.

About the FoV, isn't better to be conservative for the trigger rate calculation? We want to simulate every event that may have any chance of triggering and there may be many of them slightly outside the nominal FoV.

Similar argument for the energy. Even low energy proton showers may contribute for the trigger rate of SSTs.

Reply:
Yes, it is better to be conservative. But what about the other way around? Tomorrow we upgrade telescopes with a larger FOV, will we notice that there is a hardwired parameter? My main point is really that we shouldn't have hardwired parameters.

Reply:
It's not hardwired. This is an application, which means a specific tool that requires specific parameters.
We keep everything as flexible as possible along the library. The point of the application is to reduce all this flexibility to the minimum necessary for executing an specific task. 

And yes, if we upgrade our telescopes with a larger FoV, we will surely need to do some work to upgrade our software too.
And if the work on the simulation software means changing parameters at the application level, I consider this a win in terms of maintainability. 

On the other hand, if every time we want to calculate trigger rates we need to set FoV, energy range etc, I consider this a big loss in usability.

Reply:
Nono! I do not expect anything to be hardwired anywhere. All should be be in the configuration.

And as default configurations are provided (in the DB or through configuration files), one does not have to set the parameters each time you start it.

Whatever you hardwire now will be the reason for problems in the future.

Reply:
I will add this is a topic for discussion, but I'd keep it as it is by now.
This issue actually relates to a couple of other application as well, not only this one.